### ISSUE TITLE
"order-pay" and "view_order" endpoint url not working 

### ISSUE THREAD

**Issue by niwinnm** [woothemes/woocommerce#9877]
The **order-pay** link doesn't work. It says invalid order id. I create the order via **create_order** API endpoint and used the order id (obtained from the response JSON) to redirect the user to the payment page, https://www.example.com/checkout/order-pay/<order_id>/

It says, the order ID is invalid. I checked the dashboard as well, the order id seems to be correct. 

Similarly, **view_order** endpoint doesn't work as well. It says the same error.

**Comment by mikejolley** [comment]
2 things; 

1. These endpoints need an order ID specified in the URL. /view-order/100 /order-pay/100
2. There are owner checks, for example in view order:

`
if ( ! current_user_can( 'view_order', $order_id ) ) {
`

If your logged in user cannot view the order (wrong role,or not the owner), it won't allow you to view it and you'll get the error.

**Comment by niwinnm** [comment]
I did pass the order_id to the URLs. So that shouldn't be the issue. So the issue must related to the Owner checks. Basically am developing an android app with the WooCommerce API. While placing order, I create a new order with the items in cart and am trying to redirect the user to a browser using the view_order link or pay_order link. 

So any idea, how should I solve this owner check issue?

**Comment by mikejolley** [comment]
This won't work for you because you cannot log the user in. They won't be able to view via that page.

For pay however, just don't set the user ID: https://github.com/woothemes/woocommerce/blob/753014da5a6df243928e4b7baebb035300ab70ff/includes/wc-user-functions.php#L292

**Comment by niwinnm** [comment]
I didn't set any user_id specifically. When I sent a request to the create_order API, by default the user_id (id param inside the customer object) was set to 0. But I set user's mail id while I passed the value for shipping and billing address. Would that cause any trouble. 

I also tried sending a simple POST request for creating an order using POSTMAN and tried accessing the same order id using firefox browser. So am sure there no user logged in while am sending request as well as while am loading the pay order url (https://www.<mywebsite>.com/checkout/order-pay/<order_id>/)
